minos/common/model/dynamic/diff.py

```diff
@@ -10,90 +10,180 @@
+        kwargs["type_"] = model_type.type_hints["value"]
+        return cls(*args, **kwargs)
 
-        logger.debug(f"Computing the {cls!r} between {a!r} and {b!r}...")
-        fields = cls._diff(a.fields, b.fields)
-        for name in ignore:
-            fields.pop(name, None)
 
-        return cls(fields)
+class IncrementalDiff(Diff, Generic[T]):
+    """Incremental Diff class."""
 
-    @staticmethod
-    def _diff(a: dict[str, Field], b: dict[str, Field]) -> dict[str, Field]:
-        """Compute the difference between ``a`` and ``b``.
+    action: Action
+
+    def __init__(self, name: str, type_: Type, value: Any, action: Action):
+        Model.__init__(self, [Field("name", str, name), Field("value", type_, value), Field("action", Action, action)])
+
+
+class FieldsDiff(BucketModel):
+    """FieldsDiff class."""
+
+    def __init__(self, diffs: list[Diff] = None, fields: Union[Iterable[Field], dict[str, Field]] = None, **kwargs):
+        if diffs is not None:
+            fields = [Field(self.generate_random_str(), Diff, v) for v in diffs]
+        super().__init__(fields, **kwargs)
+
+        mapper = defaultdict(list)
+        for name, field in self._fields.items():
+            mapper[field.value.name].append(name)
+        self._name_mapper = mapper
+
+    def __getattr__(self, item: str) -> Any:
+        try:
+            return super().__getattr__(item)
+        except AttributeError as exc:
+            if item in self._name_mapper:
+                values = [getattr(self, name) for name in self._name_mapper.get(item)]
+                if len(values) == 1:
+                    return values[0]
+                return values
+            raise exc
+
+    def __eq__(self, other):
+        return type(self) == type(other) and tuple(self.values()) == tuple(other.values())
+
+    def __repr__(self) -> str:
+        fields_repr = ", ".join(f"{diff.name}={diff.value}" for diff in self.values())
+        return f"{type(self).__name__}({fields_repr})"
+
+    def values(self) -> Iterator[Union[Diff, list[IncrementalDiff]]]:
+        """Iterate over diff instances.
+
+        :return: An iterator instance.
+        """
+        return map(lambda name: getattr(self, name), self._name_mapper.keys())
 
-        The implemented approach is equivalent to `dict(set(a.items()) -`set(b.items()))` but without requesting any
-        hashing assumptions.
+    @classmethod
+    def from_difference(cls, a: Model, b: Model, ignore: Optional[set[str]] = None) -> FieldsDiff:
+        """Build a new instance from the difference between two models.
 
-        :param a: The first dictionary of fields.
-        :param b: The second dictionary of fields.
-        :return: The differences dictionary between ``a`` and ``b``.
+        :param a: Latest model instance.
+        :param b: Oldest model instance.
+        :param ignore: Set of fields to be ignored.
+        :return: A new ``FieldsDiff`` instance.
         """
+        if ignore is None:
```

Comment:
Why don't you create the set by default in the header?
```python
ignore: Optional[set[str]] = set()
```

Since `set` is not mutable I don't think there is any problem as with `list`.

Reply:
Hi @albamig, `set` is mutable, but we have the `frozenset` data structure, which is inmutable. We can use it and simplify the code. Thanks for your feedback!

Reply:
Oooops :man_cartwheeling: I was thinking of tuples instead of sets, sorry!